### PR TITLE
Feature/make generator aware of entity interface

### PR DESCRIPTION
**Pull request by lapistano**
to be able to make the security component testable and documents more reliable the ability to define a set of interfaces to be recognized when a document is created.

### CODE REVIEW

**Comment by narcoticfresh** [comment]
Hmm.. this actually contains a lot more then it says it does..
RQL Bundle changed, Auth stuff, Interface.. hard to see if it breaks nothing..

If that is merged, what actually changes in behavior as opposed to current `develop`?

File: dev-cleanstart.sh [hunk 1]
```diff
@@ -1,4 +1,7 @@
+export SYMFONY__GENERATOR__DYNAMICBUNDLES__MONGOCOLLECTION="LoadConfig"
+export SYMFONY__GENERATOR__DYNAMICBUNDLES__MONGOCOLLECTION__CRITERIA='{"loadPackage":"szkb-wcp"}'
+
```
Comment (narcoticfresh) [inline dev-cleanstart.sh]: this shouldn't be here (environment variables)

File: src/Graviton/RestBundle/Model/DocumentModel.php [hunk 2]
```diff
@@ -73,38 +73,38 @@ public function setRepository(ObjectRepository $countries)
...
         if (!empty($filter)) {
             // set filtering attributes on request
             $request->attributes->set('filtering', true);
 
-            $queryParser = new Query(urldecode($filter));
-            $queriable = new MongoOdm($this->repository, $numberPerPage, $startAt);
-            $queriable = $queryParser->applyToQueriable($queriable);
-            $records = $queriable->getDocuments();
+            // define offset
+            $queryBuilder->skip($startAt);
```
Comment (narcoticfresh) [inline src/Graviton/RestBundle/Model/DocumentModel.php]: It seems, `queryBuilder->skip()` only gets called when a filter is present -> should be called either way..

File: src/Graviton/CoreBundle/Document/GravitonShowcaseInterface.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    public function getContact();
+    public function setContact(\Graviton\PersonBundle\Document\PersonContact $contact);
+    public function getContacts();
+    public function setContacts($contacts);
+    public function getContactcode();
+    public function setContactcode(\GravitonDyn\ShowcaseBundle\Document\ShowcaseContactCode $contactCode);
+    public function getNestedarray();
+    public function setNestedarray($array);
+    public function getUnstructedobject();
+    public function setUnstructedobject($object);
```
Comment (narcoticfresh) [inline src/Graviton/CoreBundle/Document/GravitonShowcaseInterface.php]: this interface shouldn't be here..

**Comment by narcoticfresh** [comment]
As said - IMHO most OK (even if it mixes a lot). We'll leave that PR on hold as for now, so we can merge everything except that concerning the Interface problems of generated Documents. That we'll need to solve separately. As said, Showcase Entities should always be present in the PHPUnit run so they should be testable without Interfaces.

Once Airlock is in a seperate bundle; we can test that stuff in the wrapper, where all necessary bundles are in place and available..

**Comment by lapistano** [comment]
this one is too bis .. will be spit

**Pull request closed without merging**